examples/webgpu_compute_sort_bitonic.html

```diff
@@ -130,125 +122,333 @@
+				const animate = () => {
 
-					return idx;
+					renderer.render( scene, camera );
 
 				};
 
-				const localStorage = workgroupArray( 'uint', 64 * 2 );
+				renderer.setAnimationLoop( animate );
+				setupDomElement( renderer );
+				scene.background = new THREE.Color( 0x313131 );
+
+				const bitonicSortModule = new BitonicSort(
+					renderer,
+					currentElementsStorage,
+					{
+						workgroupSize: 64,
+						sideEffectBuffers: [ 2, 3 ]
+					}
+				);
 
-				// Swap the elements in local storage
-				const localCompareAndSwap = ( idxBefore, idxAfter ) => {
+				scene.add( createDisplayMesh2( currentElementsStorage, bitonicSortModule.infoStorage ) );
 
-					If( localStorage.element( idxAfter ).lessThan( localStorage.element( idxBefore ) ), () => {
+				// Initialize each value in the elements buffer.
+				const computeInit = computeInitFn().compute( size );
+				const computeReset = computeResetBuffersFn().compute( size );
 
-						atomicAdd( counterStorage.element( 0 ), 1 );
-						const temp = localStorage.element( idxBefore ).toVar();
-						localStorage.element( idxBefore ).assign( localStorage.element( idxAfter ) );
-						localStorage.element( idxAfter ).assign( temp );
+				await renderer.computeAsync( computeInit );
 
-					} );
+				gui.add( effectController, 'stepBitonic' ).onChange( async () => {
+
+					if ( currentStep < bitonicSortModule.stepCount ) {
+
+						await bitonicSortModule.computeStep( renderer );
+
+						currentStep ++;
+
+					} else {
+
+						await renderer.computeAsync( computeReset );
+
+						currentStep = 0;
+
+					}
+
+					timestamps[ 'local_swap' ].innerHTML = constructInnerHTML( false, localColors );
+
+				} );
+
+
+
+				 const stepAnimation = async function () {
+
+					renderer.info.reset();
+
+					if ( currentStep < bitonicSortModule.stepCount ) {
+
+						bitonicSortModule.computeStep( renderer );
+
+						currentStep ++;
+
+					} else {
+
+						renderer.compute( computeReset );
+
+						currentStep = 0;
+
+					}
+
+					timestamps[ 'local_swap' ].innerHTML = constructInnerHTML( false, localColors );
+
+					if ( currentStep === bitonicSortModule.stepCount ) {
+
+						setTimeout( stepAnimation, 1000 );
+
+					} else {
+
+						setTimeout( stepAnimation, 100 );
+
+					}
 
 				};
 
+				stepAnimation();
+
+				window.addEventListener( 'resize', onWindowResize );
+
+				function onWindowResize() {
+
+					windowResizeCallback( renderer, scene, camera );
+
+				}
+
+			}
+
+			initBitonicSort();
+
+			// Global Swaps Only
+			initGlobalSwapOnly();
+
+			// When forceGlobalSwap is true, force all valid local swaps to be global swaps.
+			async function initGlobalSwapOnly() {
+
+				let currentStep = 0;
+
+				const aspect = ( window.innerWidth / 2 ) / window.innerHeight;
+				const camera = new THREE.OrthographicCamera( - aspect, aspect, 1, - 1, 0, 2 );
+				camera.position.z = 1;
+
+				const scene = new THREE.Scene();
+
+				const infoArray = new Uint32Array( 3, 2, 2 );
```

Comment:
Should this be `new Uint32Array( [ 3, 2, 2 ] )`? Not sure if it matters.

Reply:
Good catch! This line should definitely be updated. Typed arrays only accept a single scalar which defines their length. The current code ends up like so:

<img width="951" height="50" alt="image" src="https://github.com/user-attachments/assets/15a81800-dd47-4064-85a0-f521b0407be1" />